antelope_contracts/contracts/erc20/include/erc20/erc20.hpp

```diff
@@ -3,144 +3,117 @@
+            return id;
+        }
+        EOSLIB_SERIALIZE(impl_contract_t, (id)(address));
+    };
+    typedef eosio::multi_index<"implcontract"_n, impl_contract_t> impl_contract_table_t;
+
+    struct [[eosio::table("tokens")]] token_t {
+        uint64_t id = 0;
+        eosio::name token_contract;
+        bytes address;  // <-- proxy contract addr
+        eosio::asset ingress_fee;
+        eosio::asset balance;  // <-- total amount in EVM side
+        eosio::asset fee_balance;
+        uint8_t erc20_precision = 0;
+
+        uint64_t primary_key() const {
+            return id;
+        }
+        uint128_t by_contract_symbol() const {
+            uint128_t v = token_contract.value;
+            v <<= 64;
+            v |= ingress_fee.symbol.code().raw();
+            return v;
+        }
+        checksum256 by_address() const {
+            return make_key(address);
+        }
+
+        EOSLIB_SERIALIZE(token_t, (id)(token_contract)(address)(ingress_fee)(balance)(fee_balance)(erc20_precision));
+    };
+    typedef eosio::multi_index<"tokens"_n, token_t,
+                               indexed_by<"by.symbol"_n, const_mem_fun<token_t, uint128_t, &token_t::by_contract_symbol> >,
+                               indexed_by<"by.address"_n, const_mem_fun<token_t, checksum256, &token_t::by_address> > >
+        token_table_t;
+
+    struct [[eosio::table("egresslist")]] allowed_egress_account {
+        eosio::name account;
+
+        uint64_t primary_key() const { return account.value; }
+        EOSLIB_SERIALIZE(allowed_egress_account, (account));
+    };
+    typedef eosio::multi_index<"egresslist"_n, allowed_egress_account> egresslist_table_t;
+
+    uint64_t get_next_nonce();
     void handle_erc20_transfer(const token_t &token, eosio::asset quantity, const std::string &memo);
 
-    // actions defined in evm_runtime contract
     void call(eosio::name from, const bytes &to, const bytes& value, const bytes &data, uint64_t gas_limit);
-    using call_action = action_wrapper<"call"_n, &erc20::call>;
-
-    void assertnonce(eosio::name account, uint64_t next_nonce);
-    using assertnonce_action = action_wrapper<"assertnonce"_n, &erc20::assertnonce>;
-
-
-   struct  message_receiver {
 
-      enum flag : uint32_t {
-         FORCE_ATOMIC = 0x1
-      };
-
-      eosio::name     account;
-      eosio::name     handler;
-      eosio::asset    min_fee;
-      uint32_t flags;
-
-      uint64_t primary_key() const { return account.value; }
-      bool has_flag(flag f) const {
-         return (flags & f) != 0;
-      }
-
-      EOSLIB_SERIALIZE(message_receiver, (account)(handler)(min_fee)(flags));
-   };
-
-typedef eosio::multi_index<"msgreceiver"_n, message_receiver> message_receiver_table;
 
+    void assertnonce(eosio::name account, uint64_t next_nonce);
+    
 
 };
+using call_action = action_wrapper<"call"_n, &erc20::call>;
+using assertnonce_action = action_wrapper<"assertnonce"_n, &erc20::assertnonce>;
```

Comment:
You should be able to remove `assertnonce`, `call`, `call_action`, `assertnonce_action` from this file since you already have captured them in the new `evm_runtime.hpp` file.

Reply:
yeah, I meant to do that, but there's some unsaved files... fixed